### PR TITLE
PAPP-21533, PAPP-23418:   Updated pdfminer library and exception handling

### PR DESCRIPTION
**Pull request by amangal-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2021 Splunk Inc.
+   Copyright (c) 2017-2022 Splunk Inc.
```
Comment (paulnawat) [inline LICENSE:189]: Just wanna share that going forward you only need to update the license year in the App json https://github.com/splunk-soar-connectors/parser/blob/858e535ab1e367def800545988075c3405dd4e3a/parser.json#L14

Other license info will be automatically updated by the `start-release` action

File: parser_methods.py [hunk 2]
```diff
@@ -404,7 +405,7 @@ def pdf_xref_objects_to_xml(cls, pdf_file):
         text = ''
         with open(pdf_file, 'rb') as fp:
             parser = PDFParser(fp)
-            doc = PDFDocument(parser, None)
+            doc = PDFDocument(parser)
```
Comment (paulnawat) [inline parser_methods.py:408]: Is this b/c None is passed as the 2nd param by default?

File: parser.json [hunk 3]
```diff
@@ -50,15 +37,15 @@
...
+                "input_file": "wheels/defusedxml-0.7.1-py2.py3-none-any.whl"
             },
             {
                 "module": "idna",
                 "input_file": "wheels/shared/idna-2.10-py2.py3-none-any.whl"
             },
             {
                 "module": "pdfminer",
-                "input_file": "wheels/py3/pdfminer-20191125-py3-none-any.whl"
+                "input_file": "wheels/pdfminer.six-20211012-py3-none-any.whl"
```
Comment (paulnawat) [inline parser.json:48]: Could you run `pre-commit run --all-files` as well?
I wanna make sure if the path here is passing on the script. Somehow, I remember it adds wheels files under `shared`. Thanks!

**Review by paulnawat — COMMENTED** [review]
Changes look great. Could you help answer the 2 questions I left?

File: parser_methods.py (hunk 2 above)
Comment (paulnawat) [reply parser_methods.py:408]: From [their docs](https://pdfminersix.readthedocs.io/en/latest/tutorial/composable.html), this looks correct. I'm not sure why `None` was needed before. 🤔

**Review by paulnawat — APPROVED** [review]
Seeing the pipeline is green and the path should be easily verified with `pre-commit run --all-files`, I am approving this CL.

File: LICENSE (hunk 1 above)
Comment (amangal-crest) [reply LICENSE:189]: Thank you.

File: parser.json [hunk 4]
```diff
@@ -18,71 +18,70 @@
...
-                "module": "defusedxml",
-                "input_file": "wheels/defusedxml-0.7.1-py2.py3-none-any.whl"
-            },
-            {
-                "module": "pdfminer",
-                "input_file": "wheels/pdfminer-20140328-py2-none-any.whl"
-            }
-        ]
-    },
-    "pip3_dependencies": {
```
Comment (paulnawat) [inline parser.json:37]: Are these module changes done by running the `pre-commit run --all-files`?

Comment (amangal-crest) [reply parser.json:37]: yes, these changes are by running the pre-commit hook. Can you please re-review it and merge?

**Pull request merged**